test/TestVectorFieldCalc.cc

```diff
@@ -0,0 +1,131 @@
+using ::testing::FloatEq;
+using ::testing::FloatNear;
+
+std::unordered_map<CARTA::PolarizationType, float> stokes_test_values_map{
+    {CARTA::PolarizationType::POLARIZATION_TYPE_NONE, 1}, // sqrt(3*3+4*4)
+    {CARTA::PolarizationType::I, 2}, {CARTA::PolarizationType::Q, 3}, {CARTA::PolarizationType::U, 4}, {CARTA::PolarizationType::V, 5}};
+
+CARTA::SetVectorOverlayParameters SourceTestMessage(int intensity = VectorFieldCalculator::COMPUTED,
+    int angle = VectorFieldCalculator::COMPUTED, bool fractional = false, int u_error = 0, int q_error = 0) {
+    CARTA::SetVectorOverlayParameters message;
+    message.set_stokes_intensity(intensity);
+    message.set_stokes_angle(angle);
+    message.set_fractional(fractional);
+    message.set_debiasing(u_error && q_error);
+    message.set_u_error(u_error);
+    message.set_q_error(q_error);
+    message.set_smoothing_factor(1); // 1 = no downsampling
+
+    // TODO : when set to 1e20 or 1000  -> FillTileData fills everything with NaNs !!!??? std::nan here does not compile
+    message.set_threshold(std::numeric_limits<double>::quiet_NaN()); // disable threshold , std::nan does not compile !!!
+
+    return message;
+}
+
+using TestParameters = std::tuple<CARTA::SetVectorOverlayParameters, bool, float, float>;
+
+// Define the parameterized test fixture
+class VectorFieldCalcParamTest : public ::testing::TestWithParam<TestParameters> {
+public:
+    VectorFieldCalcParamTest() {}
+
+protected:
+    // You can add setup/teardown logic here if needed
+};
+
+TEST_P(VectorFieldCalcParamTest, TestStokes) {
+    // the reason to have local variable is to not capture global variables in lambda expressions (only local variables)
+    // as this does not compile on MacOS (fails CI/CD on github)
+    std::unordered_map<CARTA::PolarizationType, float> stokes_test_values_map_local = stokes_test_values_map;
+    auto [test_parameters, has_stokes_axis, expected_intensity, expected_angle] = GetParam();
+
+    // TODO : to be added as one more parameter:
+    // bool has_stokes_axis = true; // (test_parameters.stokes_angle() > 0);
+    VectorFieldCalculator vectorfield(test_parameters, has_stokes_axis);
+    std::vector<CARTA::VectorOverlayTileData> actual_intensities, actual_angles;
+
+    // lambda expression receiving tile data (messege as sent to front-end) and checking if all values = 1 (as expected for Stokes I)
+    auto callback = [&actual_intensities, &actual_angles](CARTA::VectorOverlayTileData& message) {
+        // std::cout << "DEBUG : received a message intensity tile size = " << message.intensity_tiles_size() << " , angle tile size = " <<
+        // message.angle_tiles_size() << std::endl;
+        EXPECT_EQ(message.intensity_tiles_size(), 1);
+        EXPECT_EQ(message.angle_tiles_size(), 1);
+
+        // copy messages :
+        actual_intensities.push_back(message);
+
+        if (message.angle_tiles_size() > 0) {
+            actual_angles.push_back(message);
+        }
```

Comment:
You should have a single `messages` vector, and just push the message onto the vector once.